x/subkeys/internal/types/tx/tx.go

```diff
@@ -0,0 +1,241 @@
+package tx
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/multisig"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/exported"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+var (
+	_ sdk.Tx = DelegatedTx{}
+
+	maxGasWanted = uint64((1 << 63) - 1)
+)
+
+// DelegatedTx is wraps a Msg with Fee and Signatures,
+// adding the ability to delegate the fee payment
+// NOTE: the first signature responsible for paying fees, either directly,
+// or must be authorized to spend from the provided Fee.FeeAccount
+type DelegatedTx struct {
+	Msgs       []sdk.Msg                `json:"msg" yaml:"msg"`
```

Comment:
Since I change one field, not adding one, this seems somewhat difficult.

If I just added a field, I would agree with your suggestion of embedding it.

Reply:
I think the benefit stands regardless. If you embed it, you override the field (with the new one) and get to avoid having to repeat a lot of duplicate code, no/

Reply:
Look at this:

```go
type DelegatedTx struct {
   StdTx `json:"std_tx"`
   Fee DelegatedFee `json:"fee"`
}
```

If you json marshal that, you will end up with both the Fee on the top level and Fee on the internal one (which is generally empty). So we'd have:

```json
{
  "fees": {/* delegated*/},
  "std_tx": {
    "fees": {/*maybe empty*/,
    // ...
  }
}
```

To save copying some code that is very unlikely to change, we end up with some duplicated fields and less clear serialization format.

Reply:
you dont add a json tag to `StdTx`. Go's stdlib json will flatten the keys.